### PR TITLE
Add Custom Module Creation Script and Update README

### PR DESCRIPTION
**Pull request by amanullah7649**
This merge request introduces a custom script that automates the creation of a NestJS module along with additional folder structures (e.g., `dto`, `interfaces`, `repositories`, `helpers`). This adjustment aims to improve project structure consistency and speed up module setup.

#### **Changes:**
1. **Custom Script:**
   - Added a new script (`create-module.js`) located in the `scripts` folder.
   - This script:
     - Generates a new module using the NestJS CLI.
     - Creates a controller and service for the module.
     - Adds additional folders (`dto`, `types`, `repositories`, `helpers`) within the module directory for better organization.

2. **README Update:**
   - Updated the `README.md` file to include instructions on how to use the custom module creation script.
   - Detailed the script’s functionality and provided a usage example.

#### **How to Use:**
To create a new module with the above folder structure, run:

```bash
npm run create:module <module-name>
```

Example:
```bash
npm run create:module user
```

This will generate a module under `src/modules/<module-name>` with the appropriate structure and files.

### CODE REVIEW

**Comment by AHS12** [comment]
@amanullah7649 I like this change but I want you to follow our existing CLI script pattern, move your changes to a new cli script inside the 'commands' folder. Please follow this pattern
https://github.com/Innovix-Matrix-Systems/ims-nest-api-starter/blob/main/src/commands/xsecurity.command.ts

File: scripts/create-module.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+const { execSync } = require('child_process');
```
Comment (AHS12) [inline scripts/create-module.js:1]: move this to the `commands` folder and follow our CLI pattern

**Review by AHS12 — CHANGES_REQUESTED** [review]
please follow our exisitng CLI pattern, use the `nestjs-command` package

File: scripts/create-module.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  // Generate module, controller, and service using Nest CLI
+  runCommand(`nest g module ${modulePath}`);
+  runCommand(`nest g controller ${modulePath}`);
+  runCommand(`nest g service ${modulePath}`);
+
+  // Create additional folders inside the module
+  const basePath = path.join(__dirname, '../src', modulePath);
+  createDir(`${basePath}/dto`);
+  createDir(`${basePath}/interfaces`);
```
Comment (AHS12) [inline scripts/create-module.js:36]: we are using `types.d.ts` file for interfaces, so let's create a types.d.ts file here instead of interfaces folder

**Comment by AHS12** [comment]
@amanullah7649 Thanks for changing. We are using https://www.npmjs.com/package/nestjs-command for custom commands, so I request you use this package.
you can easily move your code and use the new format. you can follow [nestjs-command ](https://www.npmjs.com/package/nestjs-command)documentation to understand the basic setup. We have already integrated this package into our codebase, just creating another command should work fine.
thanks for contributing

File: scripts/create-module.js (hunk 1 above)
Comment (amanullah7649) [reply scripts/create-module.js]: @AHS12  I have made the changes as you specified.

File: src/commands/create-module.command.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    // Generate module, controller, and service using Nest CLI
+    this.runCommand(`nest g module ${relativePath}`);
+    this.runCommand(`nest g controller ${relativePath}`);
+    this.runCommand(`nest g service ${relativePath}`);
+
+    // Create additional folders inside the module
+    this.createDir(`${modulePath}/dto`);
+    this.createDir(`${modulePath}/repositories`);
+    this.createDir(`${modulePath}/helpers`);
```
Comment (AHS12) [inline src/commands/create-module.command.ts:33]: we have global utils folder, so we don't need the helper folder here

**Review by AHS12 — CHANGES_REQUESTED** [review]
@amanullah7649 thank you for updating your code, I have some minor request, after that we can merge it.

File: src/commands/create-module.command.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import { Command } from 'nestjs-command';
+import * as path from 'path';
+
+@Injectable()
+export class CreateModuleCommand {
+  @Command({
+    command: 'create:module <moduleName>',
+    describe: 'Create a new module with a predefined structure',
+  })
+  async run(): Promise<void> {
```
[the remark this answers is not in the document]
Comment (amanullah7649) [reply src/commands/create-module.command.ts]: @AHS12  Added @Positional for moduleName as per review feedback

[the remark this answers is not in the document]
Comment (AHS12) [reply src/commands/create-module.command.ts]: @amanullah7649 thank you so much, After final check I will merge the PR

**Comment by AHS12** [comment]
@amanullah7649 thank you for your contribution, wonderful work.

**Pull request merged**